From e4be373b1fbf9822e0d02c7d763980348ab254c9 Mon Sep 17 00:00:00 2001
From: Vignesh Raghavendra <vigneshr@ti.com>
Date: Tue, 17 Sep 2019 14:53:28 +0530
Subject: [PATCH 08/23] serial: 8250: 8250_omap: Remove redundant call to
 omap_8250_rx_dma_flush

omap_8250_rx_dma_flush() is called twice in am654_8250_handle_rx_dma()
and first call quite redundant in case of second one. Drop the redundant
call.

Reported-by: Peter Ujfalusi <peter.ujfalusi@ti.com>
Signed-off-by: Vignesh Raghavendra <vigneshr@ti.com>
---
 drivers/tty/serial/8250/8250_omap.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/tty/serial/8250/8250_omap.c b/drivers/tty/serial/8250/8250_omap.c
index 77c1a18d4f43..088232f6aa75 100644
--- a/drivers/tty/serial/8250/8250_omap.c
+++ b/drivers/tty/serial/8250/8250_omap.c
@@ -1105,7 +1105,6 @@ static unsigned char am654_8250_handle_rx_dma(struct uart_8250_port *up,
 		if (!up->dma->rx_running) {
 			omap_8250_rx_dma(up);
 		} else {
-			omap_8250_rx_dma_flush(up);
 			/*
 			 * Disable RX timeout, read IIR to clear
 			 * current timeout condition, clear EFR2 to
-- 
2.17.1

